GDJS/Runtime/runtimeobject.ts

```diff
@@ -730,6 +778,168 @@ namespace gdjs {
       return this._variables.has(name);
     }
 
+    /**
+     * Returns the collection of effects to be rendered
+     * by the underlying renderer.
+     * @returns The render effects.
+     */
+    getRendererEffects() {
+      return this._rendererEffects;
+    }
+
+    initializeEffect(effectsData: EffectData) {
```

Comment:
Why do we need both a `initializeEffect` and a `addEffect`? Can we have only one, `addEffect`?

Reply:
The reason for breaking `addEffect` into multiple functions (`initializeEffect, applyEffect`) was because the renderer was created after the constructor (of runtime object) has been called. I wasn't sure how to retain the `effectsData` outside the constructor (and if it even made any sense to retain the initial effect data as a field in the object or not). Maybe there's a better design to handle it (instead of breaking the function and calling them at different times)?

Reply:
I see, we indeed need to `applyEffect` at the very end, outside of the base RuntimeObject constructor.
In this case: 
- add a comment when calling `initializeEffect` in the constructor ("We call initializeEffect as we can't use addEffect, as the renderer is not ready yet. This means applyEffect will have to be called later (see onCreated)").
- avoid this function entirely if you can (just do it in the constructor. Because each new function is complexifying the interface of the runtime object, which is already very large.
- if you can't avoid this function, at least make it private (underscore + private keyword in TypeScript). We don't want any outside caller to mess with this - it's an implementation detail, everyone else should be calling addEffect :)

Reply:
This being said, I can't think of a better design that would allow us to scrap entirely initialize/apply effect - not sure it's possible unfortunately!
Let's just try to remove as much initializeEffect as possible to keep the new interface very small.